firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java

```diff
@@ -148,6 +149,7 @@ public CollectionReference collection(@NonNull String collectionPath) {
    * @return A Task that will be resolved when the write finishes.
    */
   @NonNull
+  @NonNullGeneric
```

Comment:
Good point, ideally I would want this to have the form `Task<Unit>` but for whatever reason kotlin does not map `java.lang.Void` to `kotlin.Unit`, only the primitive `void` type.

I agree however that the fact that the only available value of type `Void` is null makes this change counterintuitive to say the least. Will change it to `@NullableGeneric`, alternatively I can just leave it unannotated and keep the type `Task<Void!>`. wdyt?

Reply:
I have no preference--I don't really know anything about kotlin's mapping mechanism. I flagged this because it didn't seem right based on my understanding of `java.lang.Void`.

I'm open to whatever works best in terms of the desired mapping. If making this `NonNullGeneric` makes for the best mapping I'm fine with that, though if possible I'd suggest we name this in a way that better suggests it's a kind of logically non-null, non-null-for-kotlin's-purposes, or non-null-in-the-kotlin-mapping.